storage/innobase/trx/trx0purge.cc

```diff
@@ -407,6 +408,20 @@ trx_purge_truncate_rseg_history(trx_rseg_t &rseg,
   {
 func_exit:
     mtr.commit();
+    if (freed && (rseg.SKIP & rseg_ref))
+    {
+      uint8_t skipped_rseg= purge_sys.m_skipped_rseg;
+      ut_ad(skipped_rseg < TRX_SYS_N_RSEGS);
+      if (&rseg == &trx_sys.rseg_array[skipped_rseg] &&
+          purge_sys.truncate.current != rseg.space)
+      {
+        rseg.clear_skip_allocation();
+        skipped_rseg++;
+        if (!(skipped_rseg%= TRX_SYS_N_RSEGS) && srv_undo_tablespaces)
```

Comment:
Can we have one function to handle the increment ? Also, it would be good to explain the second increment skipping rseg ID 0.

Reply:
Sure, we can make this an inline function of `purge_sys_t`.

Reply:
It’s better not inlined, because it is invoked rarely by the sole caller.
```c++
void purge_sys_t::rseg_enable(trx_rseg_t &rseg)
```